Boost is downloaded twice during build

CARLA version: latest dev (9037124f594daa09f745f00d7125a513cac8aaaf) 
Platform/OS: Ubuntu 22.04
Python: 3.10 & 3.7

**Problem you have experienced:**

When making a new build it currently downloads boost twice and creates two tar files.
I assume this is not a wanted behavior.

---

Output:

```sh
Util/BuildTools/Linux.mk:163: warning: overriding recipe for target 'downloadplugins'
Util/BuildTools/Linux.mk:142: warning: ignoring old recipe for target 'downloadplugins'
Setup.sh: Retrieving boost.
--2024-05-13 10:52:08--  https://archives.boost.io/release/1.80.0/source/boost_1_80_0.tar.gz
Resolving archives.boost.io (archives.boost.io)... 146.75.119.52
Connecting to archives.boost.io (archives.boost.io)|146.75.119.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 136670223 (130M) [application/octet-stream]
Saving to: ‘boost_1_80_0.tar.gz.8’

boost_1_80_0.tar.gz.8        100%[=============================================>] 130,34M  68,4MB/s    in 1,9s    

2024-05-13 10:52:10 (68,4 MB/s) - ‘boost_1_80_0.tar.gz.8’ saved [136670223/136670223]

Elapsed Time downloading from boost webpage: 2 seconds
Setup.sh: Using boost backup
--2024-05-13 10:52:11--  https://carla-releases.s3.us-east-005.backblazeb2.com/Backup/boost_1_80_0.tar.gz
Resolving carla-releases.s3.us-east-005.backblazeb2.com (carla-releases.s3.us-east-005.backblazeb2.com)... 149.137.137.254
Connecting to carla-releases.s3.us-east-005.backblazeb2.com (carla-releases.s3.us-east-005.backblazeb2.com)|149.137.137.254|:443... connected.
HTTP request sent, awaiting response... 200 
Length: 136670223 (130M) [application/x-gzip]
Saving to: ‘boost_1_80_0.tar.gz.9’

boost_1_80_0.tar.gz.9        100%[=============================================>] 130,34M  28,2MB/s    in 5,4s    

2024-05-13 10:52:18 (23,9 MB/s) - ‘boost_1_80_0.tar.gz.9’ saved [136670223/136670223]

Elapsed Time downloading from boost carla backup in backblaze: 7 seconds
```

--- a/Util/BuildTools/Setup.sh
+++ b/Util/BuildTools/Setup.sh
@@ -94,16 +94,16 @@ for PY_VERSION in ${PY_VERSION_LIST[@]} ; do
     log "Retrieving boost."
 
     start=$(date +%s)
-    wget "https://archives.boost.io/release/${BOOST_VERSION}/source/${BOOST_PACKAGE_BASENAME}.tar.gz" || true
+    wget "https://archives.boost.io/release/${BOOST_VERSION}/source/${BOOST_PACKAGE_BASENAME}.tar.gz" -O ${BOOST_PACKAGE_BASENAME}.tar.gz || true
     end=$(date +%s)
     echo "Elapsed Time downloading from boost webpage: $(($end-$start)) seconds"
 
     # try to use the backup boost we have in Jenkins
-    if [ ! -f "${BOOST_PACKAGE_BASENAME}.tar.gz" ] || [[ $(sha256sum "${BOOST_PACKAGE_BASENAME}.tar.gz") != "${BOOST_SHA256SUM}" ]] ; then
+    if [ ! -f "${BOOST_PACKAGE_BASENAME}.tar.gz" ] || [[ $(sha256sum "${BOOST_PACKAGE_BASENAME}.tar.gz" | cut -d " " -f 1 ) != "${BOOST_SHA256SUM}" ]] ; then
       log "Using boost backup"
 
       start=$(date +%s)
-      wget "https://carla-releases.s3.us-east-005.backblazeb2.com/Backup/${BOOST_PACKAGE_BASENAME}.tar.gz" || true
+      wget "https://carla-releases.s3.us-east-005.backblazeb2.com/Backup/${BOOST_PACKAGE_BASENAME}.tar.gz" -O ${BOOST_PACKAGE_BASENAME}.tar.gz || true
       end=$(date +%s)
       echo "Elapsed Time downloading from boost carla backup in backblaze: $(($end-$start)) seconds"
 

